### PR TITLE
add new Js

### PR DESCRIPTION
**Pull request by emma-luk**
add new js

### CODE REVIEW

**Comment by emma-luk** [comment]
pull requests

**Comment by emma-luk** [comment]
EmmaLuk_assignment_1

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+module.exports = {
+    "env": {
+        "browser": true
+    },
+    "extends": "eslint:recommended",
+    "rules": {
+        "indent": [
+            "error",
+            4
```
Comment (jamdon) [inline .eslintrc.js]: This value should be 2 as per the assignment requirements.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            "error",
+            "single"
+        ],
+        "semi": [
+            "error",
+            "always"
+        ],
+
+        // disallow trailing whitespace at the end of lines
+        "no-trailing-spaces": [0],
```
Comment (jamdon) [inline .eslintrc.js]: use 2 or "error" or it won't be flagged as an error by eslint.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        "no-undef-init": [0],
+
+        // disallow use of undeclared variables
+        "no-undef": [0],
+
+        // specify whether double or single quotes should be used
+        "quotes": [0],
+
+        // disallow use of console (off by default in the node environment)
+        "no-console": [0],
```
Comment (jamdon) [inline .eslintrc.js]: This should be 2 or "error" to activate this rule properly.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+        // disallow trailing commas in object literals
+        "no-comma-dangle": [0],
```
Comment (jamdon) [inline .eslintrc.js]: This should be 2 or "error" and a second configuration value of "never" is required for this rule.

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        // specify the maximum length of a line in your program (off by default)
+        "max-len": [0],
```
Comment (jamdon) [inline .eslintrc.js]: This should be 2 or "error" and additional configuration values are necessary for this rule:

`"max-len": [2, 80, 2, { ignoreUrls: true }];`

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+        ////////// Best Practices //////////
+
+        "block-scoped-var": [0],      // treat var statements as if they were block scoped (off by default)
+        "complexity": [0],            // specify the maximum cyclomatic complexity allowed in a program (off by default)
+        "consistent-return": [0],     // require return statements to either always or never specify values
+        "curly": [0],                 // specify curly brace conventions for all control statements
+        "default-case": [0],          // require default case in switch statements (off by default)
+        "dot-notation": [0],          // encourages use of dot notation whenever possible
+        "eqeqeq": [0],                // require the use of === and !==
```
Comment (jamdon) [inline .eslintrc.js]: This should be 2 or "error" to activate this rule properly.

**Comment by jamdon** [comment]
Hi Emma. It appears you still have various linting errors that haven't been corrected. I've left some comments inline with the .eslintrc file pointing out some issues with that file. If you have specific questions about anything, email me, ask in slack, or, alternatively, you can watch the homework review recording I have posted in the recordings section of the github page wiki. Let me know if you want this graded as-is, or you can also make updates, push your changes to your own forked repo, the let me know and I will review again.

**Comment by jamdon** [comment]
Hi Emma. It appears you still have linting errors. Just checking again whether your intention is to make another update or if I should just grade as-is.

**Comment by jamdon** [comment]
Hi Emma. If I don't hear back from you soon, I'm going to grade this assignment as is.

**Comment by emma-luk** [comment]
I will update this today

**Comment by jamdon** [comment]
Hi Emma. You still have linting errors. Did you see them when you ran eslint? Should I grade this as-is or are you making more updates?

**Comment by jamdon** [comment]
Assignment is graded as-is. There were 87 linting errors remaining of two varieties.

**Comment by emma-luk** [comment]
Thank you. Can I check? how to fix this?

**Comment by jamdon** [comment]
Hi Emma,

If you need more guidance on the homework, you can watch the additional
recordings I made where I go over the homework from the course github.

thanks,

James

**Pull request closed without merging**